lib/archethic.ex

```diff
@@ -189,28 +189,53 @@ defmodule Archethic do
       |> P2P.nearest_nodes()
       |> Enum.filter(&Node.locally_available?/1)
 
-    try do
-      {local_chain, paging_address} =
-        case TransactionChain.get_last_local_address(address) do
-          nil -> {[], nil}
-          last_address -> {TransactionChain.get_locally(last_address), last_address}
-        end
-
-      remote_chain =
-        if address != paging_address do
+    # We directly check if the transaction exists and retrieve the genesis
+    # Otherwise we are requesting the genesis address remotly
+    genesis_address =
+      with ^address <- TransactionChain.get_genesis_address(address),
+           {:ok, genesis_address} <- TransactionChain.fetch_genesis_address_remotely(address) do
+        genesis_address
+      else
+        _ ->
           address
-          |> TransactionChain.stream_remotely(nodes, paging_address)
-          |> Enum.to_list()
-          |> List.flatten()
+      end
+
+    %{transactions: local_chain, last_address: last_local_address} =
+      genesis_address
+      |> TransactionChain.scan_chain(address)
+      |> Enum.reduce_while(%{transactions: [], last_address: nil}, fn transaction, acc ->
```

Comment:
Do we need to reduce the chain and check the desired address, as the function `scan_chain` already stop at the desired address ?

Reply:
The reduce was mainly used to get the latest address while leveraging streams , so one item at the time. 
With List.last we would have to traverse the entire list. Also it's not supporting Stream, so we should convert it to list first. 
I didn't find yet a better way